src/DataModel/IResource.ts

```diff
@@ -0,0 +1,5 @@
+export interface IResource
+{
+    "@id": string;
+    "@type"?: Array<string>
```

Comment:
Being just pragmatic - I'd prefer to have resource's identifier as a property. Also, we'd need to extend every single JSON object from the payload with that method's implementation -  it's a drag.